tests/phpunit/tests/rest-api/rest-users-controller.php

```diff
@@ -3100,6 +3159,79 @@ public function data_get_default_data() {
 		);
 	}
 
+	/**
+	 * @ticket 56481
+	 */
+	public function test_get_item_with_head_request_should_not_prepare_user_data() {
+		wp_set_current_user( self::$user );
+		$request = new WP_REST_Request( 'HEAD', sprintf( '/wp/v2/users/%d', self::$user ) );
+
+		$hook_name = 'rest_prepare_user';
+
+		$filter   = new MockAction();
+		$callback = array( $filter, 'filter' );
+		add_filter( $hook_name, $callback );
+		$response = rest_get_server()->dispatch( $request );
+		remove_filter( $hook_name, $callback );
+
+		$this->assertSame( 200, $response->get_status(), 'The response status should be 200.' );
+
+		$this->assertSame( 0, $filter->get_call_count(), 'The "' . $hook_name . '" filter was called when it should not be for HEAD requests.' );
+		$this->assertNull( $response->get_data(), 'The server should not generate a body in response to a HEAD request.' );
+	}
+
+	/**
+	 * @dataProvider data_readable_http_methods
+	 * @ticket 56481
+	 *
+	 * @param string $method HTTP method to use.
+	 */
+	public function test_get_items_returns_only_fetches_ids_for_head_requests( $method ) {
```

Comment:
```suggestion
	public function test_get_items_returns_only_ids_for_head_requests( $method ) {
```
Suggestion per above.

Reply:
https://github.com/WordPress/wordpress-develop/pull/7970#discussion_r1876900479